src/lifecycle/props-init.js

```diff
@@ -79,6 +79,7 @@ function createNativePropertyDefinition(name, opts) {
     propData.internalValue = newValue;
 
     const changeData = { name, newValue, oldValue };
+    const valueHasChanged = newValue !== oldValue;
```

Comment:
Is it possible for a newValue to be `null` and for a oldValue to be `undefined`? (or vice versa) Don't remember at which point they are coerced. Because the strict comparison wouldn't work in this case.